vendor/engines/saas/app/controllers/saas/coupons_controller.rb

```diff
@@ -11,8 +11,8 @@ def valid
     def redeem
       query = Queries::CouchCustomer.get_cust(params[:id], couch)
       doc = QueryHandler.exec(&query)
-      return render json: {success: false, message: "Couldn't find couch record: #{plan_doc.id}"} unless doc
       plan_doc = doc
+      return render json: {success: false, message: "Couldn't find couch record: #{plan_doc.id}"} unless doc
```

Comment:
Yep, so I just moved the line down. 

could be refactored to just use doc I suppose